### PR TITLE
feat(api): :sparkles: Make mfa_input_callback available for soft-token method

### PR DESCRIPTION
**Pull request by jaron-l**
* move getpass, oathtool, and keyring hidden imports to top of file

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -19,14 +19,13 @@
     url="https://github.com/mintapi/mintapi",
     install_requires=[
         "configargparse",
-        "future",
-        "mock",
```
Comment (mrooney) [inline setup.py:23]: These ones aren't actually used?

**Comment by mrooney** [comment]
Makes sense overall. I think the original idea was that these options might not be mainstream (they obviously weren't when they were added) and that it only made sense to require them if people were going to use them. But maybe these are all popular options now or it is easier to require them all?

Also, now that we have more imports, does it make sense to alphabetize them? That can certainly be in a follow up PR.

File: setup.py (hunk 1 above)
Comment (jaron-l) [reply setup.py:23]: They're included in the standard library of python 3.

Comment (jaron-l) [reply setup.py:23]: Or rather, mock is. Future is no longer used as it was for python 2 to 3 compatibility.

**Comment by jaron-l** [comment]
> Makes sense overall. I think the original idea was that these options might not be mainstream (they obviously weren't when they were added) and that it only made sense to require them if people were going to use them. But maybe these are all popular options now or it is easier to require them all?
> 
> Also, now that we have more imports, does it make sense to alphabetize them? That can certainly be in a follow up PR.

I think it is much easier to just require them on install. Pip will just handle it then. There's really no disadvantage at all to require it on install versus having to manually install packages as you find certain features require them. Plus, the latter has no dependency management system so any advantage in not requiring would quickly get more complicated at scale. The only real optional dependency management system I've seen is in python-poetry.

I don't really see an advantage to alphabetizing them. I've been generally organizing imports by built-ins, external packages, and internal modules as that is somewhat a common practice so you know when to careful about circular imports.

**Comment by mrooney** [comment]
> I don't really see an advantage to alphabetizing them. I've been generally organizing imports by built-ins, external packages, and internal modules as that is somewhat a common practice so you know when to careful about circular imports.

That makes sense! My main desire is that if someone is looking to add or remove an import, there's a right place and not an "I'll throw it here, I guess"-place. It's a lot nicer to visually scan too. I can do that in a follow up PR though.

**Review by mrooney — APPROVED** [review]

**Pull request merged**